src/shared-components/button/style.ts

```diff
@@ -162,15 +162,15 @@ function parseTheme(
   }
 }
 
-type BaseButtonStylesTypes = {
+export interface BaseButtonStylesTypes {
```

Comment:
It looks like relying on interfaces instead of types requires us to export the interface, even when they aren't explicitly being used. (Key word is _explicitly_, more on that soon.) If we don't, we get the following error:

```shell
error TS4023: Exported variable 'RoundButtonBase' has or is using name 'BaseButtonStylesTypes' from external module "/Users/michaelaltamirano/radiance-ui/src/shared-components/button/style" but cannot be named.

export const RoundButtonBase = styled(ButtonBase)`
```

Looking at the type output, when we use `type` we are re-defining the type in-line:

<img width="624" alt="Screen Shot 2021-02-10 at 5 59 51 PM" src="https://user-images.githubusercontent.com/13544620/107588685-e9cc2200-6bc9-11eb-9e68-753910b7bdbc.png">

(This is one of the keys for why preferring `interface` results in a much smaller declaration filesize.)

When we use interface, and export it, the type is this simple instead: `Pick<Pick<import("../../style").BaseButtonStylesTypes...` 🙂